src/Kernel/Context.class.st

```diff
@@ -973,9 +981,23 @@ Context >> isContext [
 
 { #category : #testing }
 Context >> isDead [
-	"Has self finished"
+	"Context is dead when
+	- it is self finished (returned to sender) 
+	- it is at the last instruction of process termination (it is unwinded and process is suspended)"
 
-	^ pc isNil
+	^ pc isNil or: [  self isEndOfProcessTermination ]
```

Comment:
It is another important change. 
I checked users of #isDead. The critical senders were already accompanied with #isEndOfProcessTermination as part of this PR to fix existing problems. For the rest of users it also makes sense to take into account this condition. And probably it even fixes some bugs.